test/framework/src/main/java/org/elasticsearch/test/BackgroundIndexer.java

```diff
@@ -163,13 +162,11 @@ public void run() {
                                 }
                                 BulkResponse bulkResponse = bulkRequest.get();
                                 for (BulkItemResponse bulkItemResponse : bulkResponse) {
-                                    if (!bulkItemResponse.isFailed()) {
+                                    if (bulkItemResponse.isFailed() == false) {
                                         boolean add = ids.add(bulkItemResponse.getId());
                                         assert add : "ID: " + bulkItemResponse.getId() + " already used";
                                     } else {
-                                        throw new ElasticsearchException("bulk request failure, id: ["
-                                                + bulkItemResponse.getFailure().getId() + "] message: "
-                                                + bulkItemResponse.getFailure().getMessage());
+                                        failures.add(bulkItemResponse.getFailure().getCause());
```

Comment:
I had to change this so that BackgroundIndexer.totalIndexedDocs() returns the effective number of docs indexed and does not only reports the first bulk item failure.